File: backend/src/main/java/com/musicmatch/backend/dto/UpdateProfileRequest.java
package com.musicmatch.backend.dto;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.musicmatch.backend.model.ExperienceLevel;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UpdateProfileRequest {
    private String biography;
    private Long cityId;
    private ExperienceLevel experienceLevel;
    private List<Long> styleIds;
    private List<InstrumentLevelRequest> instruments;
}

File: mobile/app/src/main/java/com/musicmatch/mobile/model/dto/UpdateProfileRequest.kt
package com.musicmatch.mobile.model.dto

import com.musicmatch.mobile.model.ExperienceLevel

data class UpdateProfileRequest(
    val biography: String? = null,
    val cityId: Long? = null,
    val experienceLevel: ExperienceLevel? = null,
    val styleIds: List<Long>? = null,
    val instruments: List<InstrumentLevelRequest>? = null
)